File: API/Models/Faq.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace PropertyFlipperAPI.Models
{
    public class Faq
    {
        public int FaqId { get; set; }

        [Required]
        [MaxLength(250)]
        public string Question { get; set; } = string.Empty;

        [Required]
        public string Answer { get; set; } = string.Empty;

        public int DisplayOrder { get; set; } = 0;

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    }
}

File: Flutter/lib/models/faq.dart
class Faq {
  final String faqId;
  final String question;
  final String answer;
  final int displayOrder;

  Faq({
    required this.faqId,
    required this.question,
    required this.answer,
    required this.displayOrder,
  });

  factory Faq.fromJson(Map<String, dynamic> json) {
    // Helper to parse ID fields (handle both string GUID and int legacy formats)
    String parseId(dynamic id, {String defaultValue = ''}) {
      if (id == null) return defaultValue;
      if (id is String) return id;
      if (id is int) return id.toString(); // Legacy format
      return id.toString();
    }

    return Faq(
      faqId: parseId(json['faqId']),
      question: json['question'] ?? '',
      answer: json['answer'] ?? '',
      displayOrder: json['displayOrder'] ?? 0,
    );
  }
}
